frontend/app/services/letters-service.server.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const letterSchema = z.object({
+  dateSent: z.date().optional(),
+  letterTypeCode: z.string().optional(),
+  referenceId: z.string().uuid().optional(),
+});
+
+export type LettersInfo = z.infer<typeof letterSchema>;
+
+function createLettersService() {
+  const logger = getLogger('letters-service.server');
+  const { INTEROP_API_BASE_URI } = getEnv();
+
+  /**
+   *
+   * @param { userId }
+   * @returns returns the letters based off  @param userId
+   */
+  async function getLetters(requestBody: { userId: string }) {
+    const url = `${INTEROP_API_BASE_URI}/letters?userId=${requestBody.userId}`;
+    const response = await fetch(url);
+
+    if (response.ok) return letterSchema.parse(await response.json());
+    if (response.status === 404) return null;
```

Comment:
Interop should return an empty array if no letters are found. If we get a 404 back from interop for this call, it's almost certainly a misconfiguration on our end, so we should throw.

In other words, don't add any special handling for the 404.. let your `!response.ok` check handle it.

Reply:
FIxed: 47fffce2dc8f67ae5494d7842d37f6acb35a77cc